### PR TITLE
Improve appsettings

### PR DESCRIPTION
**Pull request by glahaye**
### Motivation and Context
Make appsettings easier to use

### Description
Remove unused part, comment better, make file easier to use and tougher to accidentally abuse

### Contribution Checklist
- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [ ] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: samples/apps/copilot-chat-app/SKWebApi/appsettings.json [hunk 1]
```diff
@@ -1,48 +1,42 @@
...
-    "Domain": "[Enter the domain of your tenant, e.g. contoso.onmicrosoft.com]",
-    "TenantId": "[Enter 'common', or 'organizations' or the Tenant Id (Obtained from the Azure portal. Select 'Endpoints' from the 'App registrations' blade and use the GUID in any of the URLs), e.g. da41245a5-11b3-996c-00a8-4d99re19f292]",
-    "ClientId": "[Enter the Client Id (Application ID obtained from the Azure portal), e.g. ba74781c2-53c2-442a-97c2-3d60re42f403]",
-    "Scopes": "access_as_user",
-    "CallbackPath": "/signin-oidc"
+    "AIService": "AzureOpenAI", // Or "OpenAI" when using OpenAI directly
+    "DeploymentOrModelId": "text-embedding-ada-002", // Azure OpenAI deployment name or OpenAI model to use
+    "Endpoint": "" // Your Azure OpenAI endpoint (e.g. https://contoso.openai.azure.com) - Leave blank when using OpenAI directly
+    /* "Key": Key for your (Azure) OpenAI instance. Use user secrets, environment variable or Key Vault to populate.
+       Putting secrets in a plain-text file is a security risk and should be done as a last resort only.
```
Comment (adrianwyatt) [inline samples/apps/copilot-chat-app/SKWebApi/appsettings.json:25]: `Putting secrets in a plain-text file is unsafe and not recommended.`

File: samples/apps/copilot-chat-app/SKWebApi/appsettings.json [hunk 2]
```diff
@@ -1,48 +1,42 @@
...
-    //"Key": "[DO NOT CHECK IN - Key to your (Azure) OpenAI instance.]"
+    "AIService": "AzureOpenAI", // Or "OpenAI" when using OpenAI directly
+    "DeploymentOrModelId": "text-davinci-003", // Azure OpenAI deployment name or OpenAI model to use
+    "Endpoint": "" // Your Azure OpenAI endpoint (e.g. https://contoso.openai.azure.com) - Leave blank when using OpenAI directly
+    /* "Key": Key for your (Azure) OpenAI instance. Use user secrets, environment variable or Key Vault to populate.
+       Putting secrets in a plain-text file is a security risk and should be done as a last resort only.
+       Ex: dotnet user-secrets set "CompletionConfig:Key" "MY_COMPLETION_KEY" */
   },
+
   // Remove or comment out EmbeddingConfig if no embedding backend is required
```
Comment (adrianwyatt) [inline samples/apps/copilot-chat-app/SKWebApi/appsettings.json:18]: Please add a line stating when/why no embedding backend would be required.

File: samples/apps/copilot-chat-app/SKWebApi/appsettings.json [hunk 3]
```diff
@@ -1,48 +1,42 @@
...
     "LogLevel": {
       "Default": "Warning"
     }
   },
+
   "AllowedHosts": "*",
+
   // CORS settings
   "AllowedOrigins": [
     "http://localhost:3000"
```
Comment (adrianwyatt) [inline samples/apps/copilot-chat-app/SKWebApi/appsettings.json:42]: Could we add a comment stating that this is the address of the frontend application?

**Review by adrianwyatt — CHANGES_REQUESTED** [review]

File: samples/apps/copilot-chat-app/SKWebApi/appsettings.json [hunk 4]
```diff
@@ -1,48 +1,42 @@
 {
-  // Consider populating secrets, such as "Key" properties, from dotnet's user secrets when running locally.
-  // https://learn.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-7.0&tabs=windows#secret-manager
-  // Ex: dotnet user-secrets set "CompletionConfig:Key" "MY_SECRET_VALUE"
-  // Values in user secrets take precedence over those in this file
+  /* Consider populating secrets, such as "Key" properties, from dotnet's user secrets when running locally.
+     https://learn.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-7.0&tabs=windows#secret-manager       
+     Values in user secrets take precedence over those in this file. */
```
Comment (adrianwyatt) [inline samples/apps/copilot-chat-app/SKWebApi/appsettings.json:5]: Could we add a logging verbosity section?
```
  "Logging": {
    "LogLevel": {
      "Default": "Information",
      "Microsoft": "Warning",
      "Microsoft.Hosting.Lifetime": "Warning",
      "System.Net.Http.HttpClient": "Warning"
    }
  },
```

**Review by adrianwyatt — CHANGES_REQUESTED** [review]

File: samples/apps/copilot-chat-app/SKWebApi/appsettings.json [hunk 5]
```diff
@@ -1,48 +1,42 @@
...
-    "DeploymentOrModelId": "[Enter the Azure OpenAI deployment name or OpenAI model, e.g. 'text-davinci-003']",
-    "Endpoint": "[Enter the endpoint of your Azure OpenAI instance, e.g. https://contoso.openai.azure.com]"
-    /* Use user secrets or Key Vault - Putting secrets in a plain-text file
-       is a security risk and should be done as a last resort only. */
-    //"Key": "[DO NOT CHECK IN - Key to your (Azure) OpenAI instance.]"
+    "AIService": "AzureOpenAI", // Or "OpenAI" when using OpenAI directly
+    "DeploymentOrModelId": "text-davinci-003", // Azure OpenAI deployment name or OpenAI model to use
+    "Endpoint": "" // Your Azure OpenAI endpoint (e.g. https://contoso.openai.azure.com) - Leave blank when using OpenAI directly
+    /* "Key": Key for your (Azure) OpenAI instance. Use user secrets, environment variable or Key Vault to populate.
+       Putting secrets in a plain-text file is a security risk and should be done as a last resort only.
```
[the remark this answers is not in the document]
Comment (glahaye) [reply samples/apps/copilot-chat-app/SKWebApi/appsettings.json:14]: Changed

File: samples/apps/copilot-chat-app/SKWebApi/appsettings.json (hunk 1 above)
Comment (glahaye) [reply samples/apps/copilot-chat-app/SKWebApi/appsettings.json:25]: Changed

File: samples/apps/copilot-chat-app/SKWebApi/appsettings.json (hunk 2 above)
Comment (glahaye) [reply samples/apps/copilot-chat-app/SKWebApi/appsettings.json:18]: Done

File: samples/apps/copilot-chat-app/SKWebApi/appsettings.json (hunk 3 above)
Comment (glahaye) [reply samples/apps/copilot-chat-app/SKWebApi/appsettings.json:42]: Done

**Review by adrianwyatt — APPROVED** [review]

**Pull request merged**